### DISCUSSION TITLE
Middleware call before app.vue

### DISCUSSION THREAD

**Discussion by silencerspirit** [nuxt/nuxt#18732] [Questions] [answered]
Hi!
There are a certain number of requests in app.vue (made by the nuxtServerInit type), there is also a request for a user in which isAuthorized.

The problem is that middleware is triggered earlier than the app.vue call
<img width="346" alt="image" src="https://user-images.githubusercontent.com/22055696/216589455-40acbc84-a057-44c1-8b2d-7fb326628f3a.png">

Also make a check something like this (on the right page)
```
if (authStore.isAuthorized) {
  navigateTo('/');
}
```

But I want to make it more beautiful, through `definePageMeta({middleware: 'authenticated'})`

**Comment by TheAlexLichter** [comment, marked as answer]
You could relocate all auth-init-related code into the middleware then. Sth. like:

```ts
export default defineNuxtRouteMiddleware(async (to) => {
  // Ignore login page
  if (to.path.replace(/\/$/, '') === '/login') {
    return
  }

  // Check authentication here somehow
  const { status } = await useYourAuthComposable()

  if (status.value === 'unauthenticated') {
    console.info('Move to login')
    return navigateTo('/login')
  }

  // Init auth and fetch here if not already initialized
  const { fetchUser, user } = useUser()
  if(!user.value) {
    await fetchUser()
  }
})
```

**Reply by silencerspirit** [reply]
Thanks for the answer, I rethought my code a little. I wanted to get away from waiting for a request before executing requests from the app.vue, but I realized that in my case, an additional request on these pages is not critical.

```javascript
export default defineNuxtRouteMiddleware(async () => {
  const authStore = useAuthStore();

  if (!authStore.GET_IS_AUTHORIZED && process.server) {
    await authStore.getUser();
  }

  return !authStore.GET_IS_AUTHORIZED ? navigateTo('/') : () => {};
});
```
